fix-desktop-file-path-in-appdata.patch

```diff
@@ -0,0 +1,47 @@
+From 263d74aa56964acf69df3a78bc6a2f4eb9f3370c Mon Sep 17 00:00:00 2001
+From: Lukas Holecek <hluk@email.cz>
+Date: Sun, 25 Feb 2018 15:29:05 +0100
+Subject: [PATCH] Fix desktop file path in AppData
+
+---
+ CMakeLists.txt                                     | 4 +++-
+ shared/{copyq.appdata.xml => copyq.appdata.xml.in} | 2 +-
+ 2 files changed, 4 insertions(+), 2 deletions(-)
+ rename shared/{copyq.appdata.xml => copyq.appdata.xml.in} (97%)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index f63b17e2..c69ada01 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -176,12 +176,14 @@ endif()
+ if (UNIX AND NOT APPLE)
+     install(FILES ${copyq_ICON_NORMAL} DESTINATION ${ICON_INSTALL_PREFIX} RENAME ${ICON_NAME}-normal.svg)
+     install(FILES ${copyq_ICON_BUSY}   DESTINATION ${ICON_INSTALL_PREFIX} RENAME ${ICON_NAME}-busy.svg)
+-    install(FILES ${copyq_APPDATA}     DESTINATION ${APPDATA_INSTALL_PREFIX} RENAME ${APPDATA_INSTALL_NAME})
+     install(FILES ${copyq_MANPAGE}     DESTINATION ${MANPAGE_INSTALL_PREFIX})
+ 
+     configure_file(${copyq_DESKTOP}.in ${copyq_DESKTOP})
+     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${copyq_DESKTOP} DESTINATION ${DESKTOP_INSTALL_PREFIX} RENAME ${DESKTOP_INSTALL_NAME})
+ 
++    configure_file(${copyq_APPDATA}.in ${copyq_APPDATA})
++    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${copyq_APPDATA} DESTINATION ${APPDATA_INSTALL_PREFIX} RENAME ${APPDATA_INSTALL_NAME})
++
+     foreach (copyq_ICON_EXTENT 16 22 24 32 48 64 128)
+         set(copyq_ICON_SIZE "${copyq_ICON_EXTENT}x${copyq_ICON_EXTENT}")
+         string(REPLACE "%SIZE%" "${copyq_ICON_SIZE}" copyq_ICON_TARGET_PREFIX "${ICON_INSTALL_PREFIX_TEMPLATE}")
+diff --git a/shared/copyq.appdata.xml b/shared/copyq.appdata.xml.in
+similarity index 97%
+rename from shared/copyq.appdata.xml
+rename to shared/copyq.appdata.xml.in
+index 9f55f486..18070627 100644
+--- a/shared/copyq.appdata.xml
++++ b/shared/copyq.appdata.xml.in
+@@ -1,7 +1,7 @@
+ <?xml version="1.0" encoding="utf-8"?>
+ <!-- Copyright 2018 Lukáš Holeček <hluk@email.cz> -->
+ <component type="desktop">
+-    <id>copyq.desktop</id>
++    <id>${DESKTOP_INSTALL_NAME}</id>
```

Comment:
I still strongly disagree with this approach. Your `id` is a single identifier for the project it should not be dependent on a build flag.

Reply:
Do I understand it correctly that the desktop file should be always `com.github.hluk.copyq.desktop`? Looking at `/usr/share/applications` on my system these files have usually simpler names.

Reply:
Flatpak requires a unique ID in that format yes. But the desktop file is a different, though related, issue. The appstream file should always be in the rdns style.